networkx/algorithms/isomorphism/tests/test_isomorphism.py

```diff
@@ -38,3 +47,9 @@ def test_faster_could_be_isomorphic(self):
     def test_is_isomorphic(self):
         assert iso.is_isomorphic(self.G1, self.G2)
         assert not iso.is_isomorphic(self.G1, self.G4)
+        assert iso.is_isomorphic(self.DG1, self.DG2)
+        assert not iso.is_isomorphic(self.DG1, self.DG3)
+        with pytest.raises(
+            nx.NetworkXError, match="Graphs G1 and G2 are not of the same type."
+        ):
+            iso.is_isomorphic(self.DG1, self.G1)
```

Comment:
Instead of adding new cases to the class, perhaps we can just use `to_directed` on the existing cases? It's not *exactly* the same as the `to_directed` will create graphs with edges in both directions, but that doesn't really matter for these tests. e.g. something like:

```suggestion

    def test_is_isomorphic_directed(self):
        assert iso.is_isomorphic(self.G1.to_directed(), self.G2.to_directed()
        assert not iso.is_isomorphic(self.G1.to_directed(), self.G3.to_directed())
        with pytest.raises(
            nx.NetworkXError, match="Graphs G1 and G2 are not of the same type."
        ):
            iso.is_isomorphic(self.G1.to_directed(), self.G1)
```

This is just a suggestion, not a blocker! I just have a preference for keeping the `setup_class` minimal